odoo/fields.py

```diff
@@ -2453,6 +2497,11 @@ def convert_to_record(self, value, record):
         prefetch_ids = IterableGenerator(prefetch_many2one_ids, record, self)
         return record.pool[self.comodel_name]._browse(record.env, ids, prefetch_ids)
 
+    def convert_to_record_multi(self, values, record):
+        prefetch_ids = IterableGenerator(prefetch_many2one_ids, record, self)
+        ids = tuple(unique(id_ for id_ in values if id_ is not None))
+        return record.pool[self.comodel_name]._browse(record.env, ids, prefetch_ids)
```

Comment:
I am just wondering: is `prefetch_ids` necessary?
```python
records = model.browse(ids)

# each `field.mapped` returns a recordset that will not prefetch beyond itself
records.mapped('partner_id.country_id.name')

for record in records:
    # each `field.mapped` returns a record that could prefetch more than itself
    record.mapped('partner_id.country_id.name')
```
In the first case, we could simplify: `prefetch_ids=ids`...

But I don't know whether it is worth the complexification.

Reply:
Conclusion: not for now...